RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -97,13 +97,42 @@ private void ReplaceAssignedByValParameterReferences(IModuleRewriter rewriter, s
             }
         }
 
+
         private void InsertLocalVariableDeclarationAndAssignment(IModuleRewriter rewriter, string localIdentifier)
         {
-            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine
-                + (_target.AsTypeDeclaration is ClassModuleDeclaration ? Tokens.Set + " " : string.Empty)
-                + localIdentifier + " = " + _target.IdentifierName;
+            var content = Tokens.Dim + " " + localIdentifier + " " + Tokens.As + " " + _target.AsTypeName + Environment.NewLine;
+            if (IsBaseTypeContext(_target))
+            {
+                content = content + localIdentifier + " = " + _target.IdentifierName;
+            }
+            else 
+            {
+                //All we can know is that it is not a Base type.  Let VBA determine
+                //the right way to assign the parameter.  The user can simplify it later.
+                string insertIsObjectCheck =
+@"If(IsObject({1})) Then
```

Comment:
hmm I think you've just found an interesting bug in the `ObjectVariableNotSet` inspection - a `Collection` has a default property (as does a `Range`); we explicitly exclude types that have a default property... which is the correct thing to do for e.g. a `Range` or any other object type with a default property really (another inspection is planned for those - "implicit default member assignment")... however in the case of collection types it doesn't make sense to skip them.

The default property for collection types is the `Item` parameterized getter, which can't be assigned without a parameter being specified.

[Line 30](https://github.com/rubberduck-vba/Rubberduck/blob/next/RetailCoder.VBE/Inspections/ObjectVariableNotSetInspection.cs#L30)

    item.AsTypeDeclaration == null || (!ClassModuleDeclaration.HasDefaultMember(item.AsTypeDeclaration)

I think we need to add an `IsCollection` (@comintern what do you think?) to `ClassModuleDeclaration`, that returns `true` when the type has a parameterized getter marked with a `VB_UserMemId = -4` attribute - and perhaps even an `IsImplicitMemberAssignmentSupported` property to encapsulate all the necessary conditions for this; that way the `ObjectVariableNotSet` inspection, and all other code that needs to care about this, can have simpler conditions to evaluate.